### DISCUSSION TITLE
Use renovate.json locally instead of remote when running renovate

### DISCUSSION THREAD

**Discussion by juancarlosjr97** [renovatebot/renovate#22726] [Request Help] [answered]
### How are you running Renovate?

Placeholder value, please select the correct response from the dropdown

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.115.2

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Wanted end result.

We have multiple configurations per data source, one for Node, another one for Docker, another one for Java, etc, and we have written a script to merge them all together.

However, the scripts run just before the project executes renovate, and because we have added some custom configurations the picked-up configuration in the renovate.json does not work as expected.

However, debugging, and I know this is a custom feature we have added, the renovate.json is correct locally in the project during the pipeline execution but because the pipeline runs with the renovate.json in the root of the project in the repository, it basically ignores the merge renovate.json file.

Is it possible to change this behavior to pick up the renovate.json in the current directory executing renovate?

### What you tried so far.

I have checked the documentation and I could not find anything that will use to change this behavior.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Sorry but I don't understand your question. `renovate.json` needs to be part of the repo. If you want "server side" config then it's done using `config.js` or `config.json`.

**Reply by juancarlosjr97** [reply]
Can I have a project that extends another `config.js` in addition to the one used by the renovate bot. I have a base configuration with credentials, etc in the renovate repository in our organization.

**Reply by rarkins** [reply]
Yes: https://docs.renovatebot.com/key-concepts/presets/

**Reply by juancarlosjr97** [reply]
Thank you. It is not exactly what we need as we have the credentials and want to centralize it, but thank you.

**Reply by rarkins** [reply, marked as answer]
You can either put credentials in your bot config (global config), or put them in a shared private repo to use with presets
